frontend/src/Onboarding/TransferCoursesScreen.tsx

```diff
@@ -35,253 +30,159 @@ const TitleText = styled.div`
+
+const TransferCoursesScreenComponent: React.FC<TransferCoursesScreenProps> = () => {
+  const major = useSelector((state: AppState) => getUserMajorFromState(state)!);
+  const completedRequirements = useSelector((state: AppState) =>
+    getCompletedRequirementsFromState(state)
+  );
+  const dispatch = useDispatch();
+  const setCompletedCourses = (completedCourses: ScheduleCourse[]) =>
+    dispatch(setCompletedCoursesAction(completedCourses));
+  const setTransferCourses = (transferCourses: ScheduleCourse[]) =>
+    dispatch(setTransferCoursesAction(transferCourses));
+
+  const [modalVisible, setModalVisible] = useState(false);
+  const [selectedRequirements, setSelectedRequirements] = useState<
+    IRequiredCourse[]
+  >([]);
+  const [otherCourses, setOtherCourses] = useState<IRequiredCourse[][]>([]);
+  const [completedNonTransfer] = useState(completedRequirements);
+
+  const addOtherCourses = (courses: ScheduleCourse[]) => {
     let reqCourseMap = courses.map((course: ScheduleCourse) => [
       {
         type: "COURSE",
         classId: +course.classId,
         subject: course.subject,
       } as IRequiredCourse,
     ]);
-    this.setState({
-      otherCourses: [...this.state.otherCourses, ...reqCourseMap],
-    });
-  }
+    setOtherCourses([...otherCourses, ...reqCourseMap]);
+  };
 
   /**
    * Separate transfer and non transfer courses, convert non transfer completed courses
    * into ScheduleCourses, and populate redux store.
    */
-  async onSubmit() {
-    let completedNonTransfer = this.state.completedNonTransfer.filter(
-      req => !this.state.selectedRequirements.includes(req)
+  const onSubmit = async () => {
+    const filteredCompleted = completedNonTransfer.filter(
+      req => !selectedRequirements.some(c => courseEq(c, req))
     );
-    let completed: ScheduleCourse[] = [];
-
-    for (let course of completedNonTransfer) {
-      const scheduleCourse = await fetchCourse(
-        course.subject,
-        String(course.classId)
-      );
-
-      if (scheduleCourse) {
-        completed.push(scheduleCourse);
-      }
-    }
-
-    let selectedCourses: ScheduleCourse[] = [];
-    for (let course of this.state.selectedRequirements) {
-      const scheduleCourse = await fetchCourse(
-        course.subject,
-        String(course.classId)
-      );
-
-      if (scheduleCourse) {
-        selectedCourses.push(scheduleCourse);
-      }
-    }
-
-    this.props.setCompletedCourses(completed);
-    this.props.setTransferCourses(selectedCourses);
-  }
-
-  /**
-   * Handles a class when it has been checked off. If it is being unchecked, it removes it,
-   * and if it is being checked, it converts it to a ScheduleCourse and adds it to selected courses
-   * @param e
-   * @param course
-   */
-  async onChecked(e: any, course: IRequiredCourse) {
-    const checked = e.target.checked;
-
-    if (checked) {
-      this.setState({
-        selectedRequirements: [...this.state.selectedRequirements, course],
```

Comment:
bug # 3: this was used to set state multiple times for a single checkbox (ie checking CS2510 and CS2511), and so one of them was getting wiped out (only setting the second of the two, CS2511, as checked).